base/src/types.rs

```diff
@@ -533,6 +566,17 @@ impl<Id, T> Type<Id, T>
     }
 }
 
+pub trait TypeRef: Sized {
```

Comment:
Why do we need an extra trait for this? Could we not just implement this method directly on `Type`?